Gems/ScriptCanvas/Code/Editor/View/Windows/Tools/UpgradeTool/Scanner.cpp

```diff
@@ -130,12 +135,16 @@ namespace ScriptCanvasEditor
             }
             else
             {
-                if (auto asset = LoadAsset(); asset.IsGraphValid())
+                auto sourceAndIsXML = LoadSource();
+                auto asset = sourceAndIsXML.first;
+                const bool isXML = sourceAndIsXML.second;
+
+                if (asset.IsGraphValid())
                 {
                     VE_LOG("Scanner: Loaded: %s ", ModCurrentAsset().RelativePath().c_str());
-                    FilterAsset(asset);
+                    FilterAsset(asset, isXML);
                 }
-                else
+                else if (!m_config.onlyIncludeLegacyXML || isXML)
```

Comment:
what about the else case?

Reply:
If the user isn't looking for xml, or if the file WAS xml, this case works as before